geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLTestUtils.scala

```diff
@@ -0,0 +1,55 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.spark
+
+import java.io.File
+import java.util.{Map => JMap}
+
+import com.vividsolutions.jts.geom.Coordinate
+import org.apache.spark.sql.SparkSession
+import org.geotools.data.simple.SimpleFeatureStore
+import org.geotools.data.{DataStore, DataUtilities}
+import org.geotools.geometry.jts.JTSFactoryFinder
+import org.joda.time.format.ISODateTimeFormat
+import org.locationtech.geomesa.features.ScalaSimpleFeature
+import org.locationtech.geomesa.utils.geotools.{GeneralShapefileIngest, SimpleFeatureTypes}
+
+import scala.collection.JavaConversions._
+
+object SparkSQLTestUtils {
+  def createSparkSession(): SparkSession = {
+    SparkSession.builder()
+      .appName("testSpark")
+      .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .config("spark.kryo.registrator", classOf[GeoMesaSparkKryoRegistrator].getName)
+      .config("spark.sql.crossJoin.enabled", "true")
+      .master("local[*]")
+      .getOrCreate()
+  }
+
+  def ingestChicago(ds: DataStore): Unit = {
```

Comment:
Yes....  the obvious options are
1.  Pull test dataset generation code out into a separate module that can be test dependency.
2.  Create a tests-jar from which module is created first and have the second depend on it.

Option 2 mess with skipping tests (I think -DskipTests maybe work, but -Dmaven.test.skip=true doesn't).  The first option takes some time and consideration, but would likely make GeoMesa a better project.  Instead of cooking up new datasets and filters for each test, we could re-use them at will.

That's my long-term plan; short term, we'll roll with a little WET code:).